### PR TITLE
US-55 | Add static code analyzer in the build workflow

### PR DESCRIPTION
**Pull request by yetanotheranand**
# User Story

https://tree.taiga.io/project/aanthir1-ser516-spring23-trinity/us/55

# Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# Change Description

See the  linked Taiga user story

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by vinaythallam — DISMISSED** [review]
The static build analyzer for the build work flow is working perfectly. Approving the request to merge

File: sbs-driver/ui/src/app/services/api.service.ts [hunk 1]
```diff
@@ -1,25 +1,22 @@
...
-  public post(url: string, data: any, options?: any) { 
-  return this.http.post(url, data, options); 
-  } 
-  public put(url: string, data: any, options?: any) { 
-  return this.http.put(url, data, options); 
-  } 
-  public delete(url: string, options?: any) { 
-  return this.http.delete(url, options); 
-  } 
+  public get(url: string, options?: any) {
```
[the remark this answers is not in the document]
Comment (yetanotheranand) [reply sbs-driver/ui/src/app/services/api.service.ts:10]: This is part of sbs-driver's team tech debt which they will fix as part of their changes.

File: sbs-driver/ui/src/app/services/api.service.ts [hunk 2]
```diff
@@ -1,25 +1,22 @@
...
+    return this.http.get(url, options);
+  }
+  public post(url: string, data: any, options?: any) {
```
[the remark this answers is not in the document]
Comment (yetanotheranand) [reply sbs-driver/ui/src/app/services/api.service.ts:13]: This is part of sbs-driver's team tech debt which they will fix as part of their changes.

File: sbs-driver/ui/src/app/services/api.service.ts [hunk 3]
```diff
@@ -1,25 +1,22 @@
...
+    return this.http.post(url, data, options);
+  }
+  public put(url: string, data: any, options?: any) {
```
[the remark this answers is not in the document]
Comment (yetanotheranand) [reply sbs-driver/ui/src/app/services/api.service.ts:16]: This is part of sbs-driver's team tech debt which they will fix as part of their changes.

File: sbs-driver/ui/src/app/services/api.service.ts [hunk 4]
```diff
@@ -1,25 +1,22 @@
...
+    return this.http.put(url, data, options);
+  }
+  public delete(url: string, options?: any) {
```
[the remark this answers is not in the document]
Comment (yetanotheranand) [reply sbs-driver/ui/src/app/services/api.service.ts:19]: This is part of sbs-driver's team tech debt which they will fix as part of their changes.

File: sbs-driver/ui/src/app/services/api.service.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (yetanotheranand) [reply sbs-driver/ui/src/app/services/api.service.ts:16]: This is part of sbs-driver's team tech debt which they will fix as part of their changes.

**Review by vinaythallam — APPROVED** [review]
All the conflicts are resolved and all the test cases are passed , can merge into main . Approving the request

**Review by atulprakash18 — APPROVED** [review]
Looks good to me. I assume that the conflicts are resolved.

**Pull request merged**